src/CommunityToolkit.Maui/Converters/MathExpressionConverter/MathExpression.shared.cs

```diff
@@ -140,7 +140,7 @@ IEnumerable<string> GetReversePolishNotation(string expression)
 		var output = new List<string>();
 		var stack = new Stack<(string Name, MathOperatorPrecedence Precedence)>();
 
-		foreach (Match? match in matches)
+		foreach (var match in matches.Cast<Match>())
```

Comment:
Does Cast throw exception or return null?

Reply:
It throws an `InvalidCastException` if the cast is invalid (and `ArgumentNullException` if `matches` was `null`, which it isn't).

`.Cast<T>()` basically just does an `as` cast on the entire `IEnumerable`:
```cs
IEnumerable<T> typedSource = source as IEnumerable<T>;
```

https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System.Core/System/Linq/Enumerable.cs#L1031-L1036